src/app/modules/ix-table2/api-data-provider.ts

```diff
@@ -0,0 +1,166 @@
+import { map, switchMap } from 'rxjs';
+import { EmptyType } from 'app/enums/empty-type.enum';
+import { ApiCallMethod } from 'app/interfaces/api/api-call-directory.interface';
+import { BaseDataProvider, paginateFn, sortFn } from 'app/modules/ix-table2/base-data-provider';
+import { SortDirection } from 'app/modules/ix-table2/enums/sort-direction.enum';
+import { TablePagination } from 'app/modules/ix-table2/interfaces/table-pagination.interface';
+import { TableSort } from 'app/modules/ix-table2/interfaces/table-sort.interface';
+import { WebSocketService } from 'app/services/ws.service';
+
+interface ApiCallParams { [key: string]: unknown }
+
+interface PaginationStrategy {
+  getParams(pagination: TablePagination): ApiCallParams;
+  paginate<T>(rows: T[], pagination: TablePagination): T[];
+  handleCurrentPage(loadFn: () => void, skipLoadFn: () => void): void;
+}
+
+interface SortingStrategy {
+  getParams<T>(sorting: TableSort<T>): ApiCallParams;
+  sort<T>(rows: T[], sorting: TableSort<T>): T[];
+  handleCurrentPage(loadFn: () => void, skipLoadFn: () => void): void;
+}
+
+export class PaginationClientSide implements PaginationStrategy {
+  getParams(): ApiCallParams {
+    return {};
+  }
+
+  paginate<T>(rows: T[], pagination: TablePagination): T[] {
+    return paginateFn(rows, pagination);
+  }
+
+  handleCurrentPage(loadFn: () => void, skipLoadFn: () => void): void {
+    skipLoadFn();
```

Comment:
Can you explain what `skipLoadFn` is and why we need it?

Reply:
The purpose of arguments of `handleCurrentPage`:
-  to give the sorting strategy a chance to decide either to do full reload on sorting (for server-side), or just sort existing rows (for client-side), based on polymorphism  to avoid `if`-branching
- same for pagination strategy

I've now improved the naming of these 2 arguments.